### PR TITLE
ci: check semver compat

### PR DESCRIPTION
**Pull request by cpu**
Use `cargo-semver-checks-action` in CI to try and catch semver breaking changes being made without the correct `Cargo.toml` version update.

Note: not perfect - this check passing is necessary but not sufficient for maintaining semver.

### CODE REVIEW

**Review by djc — APPROVED** [review]
Thanks!

**Comment by cpu** [comment]
I'm going to port our usual branch protection setup to this repo after this lands. Looks like it's a bit loosey goosey right now.

**Comment by cpu** [comment]
> I'm going to port our usual branch protection setup to this repo after this lands

Done. Updated to:

* Require PR approval
* Require CI status checks pass
* Require using the merge queue with the rebase merge strategy.

Administrative override is still permitted. It's pretty 1:1 with rustls/rustls.

**Pull request merged**
